src/scanspec/specs.py

```diff
@@ -582,16 +560,16 @@ class Spiral(Spec[Axis]):
         spec = Spiral("x", "y", 1, 5, 10, 50, 30)
     """
 
-    x_axis: A[Axis, schema(description="An identifier for what to move for x")]
-    y_axis: A[Axis, schema(description="An identifier for what to move for y")]
-    x_start: A[float, schema(description="x centre of the spiral")]
-    y_start: A[float, schema(description="y centre of the spiral")]
-    x_range: A[float, schema(description="x width of the spiral")]
-    y_range: A[float, schema(description="y width of the spiral")]
-    num: ANum
-    rotate: A[
-        float, schema(description="How much to rotate the angle of the spiral")
-    ] = 0.0
+    x_axis: Axis = Field(description="An identifier for what to move for x")
+    y_axis: Axis = Field(description="An identifier for what to move for y")
+    x_start: float = Field(description="x centre of the spiral")
+    y_start: float = Field(description="y centre of the spiral")
+    x_range: float = Field(description="x width of the spiral")
+    y_range: float = Field(description="y width of the spiral")
+    num: int = Field(min=1, description="Number of frames to produce")
```

Comment:
Can we share Field defs in multiple classes?

Reply:
Yes, but it saves significantly less code than before because you can't specify different defaults, before you had
```python
ANum = A[int, schema(min=1, description="Number of frames to produce")]

class Foo:
    num: ANum

class Bar:
    num: ANum = 1
```

Whereas now we need
```python
NumFrames = Field(min=1, description="Number of frames to produce")
NumFramesDefault1 = Field(min=1, description="Number of frames to produce", default=1)

class Foo:
    num: NumFrames

class Bar:
    num: NumFramesDefault1
```

Because the default is defined in the same place as everything else.

Reply:
Doesn't pydantic support `Annotated[int, Field(...)]`?

Reply:
Don't think so:
```shell
Traceback (most recent call last):
  File "/home/vid18871/projects/scanspec/tests/test_specs.py", line 7, in <module>
    from scanspec.specs import (
  File "/home/vid18871/projects/scanspec/src/scanspec/specs.py", line 140, in <module>
    class Repeat(Spec[Axis]):
  File "pydantic/dataclasses.py", line 224, in pydantic.dataclasses.dataclass.wrap
  File "pydantic/dataclasses.py", line 342, in pydantic.dataclasses._add_pydantic_validation_attributes
  File "pydantic/dataclasses.py", line 397, in pydantic.dataclasses.create_pydantic_model_from_dataclass
  File "pydantic/main.py", line 1027, in pydantic.main.create_model
  File "pydantic/main.py", line 198, in pydantic.main.ModelMetaclass.__new__
  File "pydantic/fields.py", line 497, in pydantic.fields.ModelField.infer
  File "pydantic/fields.py", line 476, in pydantic.fields.ModelField._get_field_info
ValueError: cannot specify `Annotated` and value `Field`s together for 'num'
```

Reply:
Caused by https://github.com/pydantic/pydantic/issues/3496
Have added TODO for when it's fixed